### ISSUE TITLE
MergeWith cobertura.xml  make error 

### ISSUE THREAD

**Issue by tiber76** [tonerdo/coverlet#262]
Hi,

I need to use /p:MergeWith  to merge differents results with dotnet test with coverlet.  

I use **coverlet.msbuild : 2.3.2**  ( but i have the same error with the last version 2.4.0 )

My command : 
dotnet test "C:\Users\Tiber\source\repos\LH.api\LH.Common\Tests\LH.Common.Repository.UnitTest\LH.Common.Repository.UnitTest.csproj" /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura /p:CoverletOutput="C:\Users\Tiber\source\repos\LH.api\LH.Common\TestResults\CodeCoverage\" /p:MergeWith="C:\Users\Tiber\source\repos\LH.api\LH.Common\TestResults\CodeCoverage\coverage.cobertura.xml" --configuration Release

My First test is done without error and generate result file here : 
C:\Users\Tiber\source\repos\LH.api\LH.Common\TestResults\CodeCoverage\coverage.cobertura.xml

But my second test ( to merge with the first ) fail with the same command, see error below :
 
![image](https://user-images.githubusercontent.com/15739525/49743530-486c7d00-fc9b-11e8-9d92-6b3dd11d7afe.png)

Thank's for support.

Jérémy

**Comment by vlef** [comment]
The value given to /p:MergeWith must be a path to Coverlet's own json result format.

There hasn't been a release of Coverlet since this feature was added, so for now you'd have to merge all of your results as json, and then convert it to the format you want.

**Comment by tiber76** [comment]
Humm ok thank's !   But how to convert immediatly this json to cobertura.xml ? do i must to convert this file manually ?

**Comment by tonerdo** [comment]
Hi @tiber76, like @vlef said, you can only merge with Coverlet's own format. Your first run should generate the regular Coverlet json and your second run should merge with the json output of the previous run and set your output to cobertura

**Comment by tiber76** [comment]
Thank you Very much :)    And Thank for your tool  👍

**Comment by farzadmf** [comment]
For future visitors, [this SO answer](https://stackoverflow.com/a/56281108/3646201) was very useful, which basically removes the redundant step of running the tests for a second time.
Basically, the solution is to output two formats, something like this in bash:
```bash
dotnet test <solution-file> \
  # ...
  /p:CoverletOutputFormat=\"json,cobertura\" \
  /p:MergeWith=<result-dir>/coverage/coverage.json
```

**Comment by MarcoRossignoli** [comment]
@farzadmf it works only if you run test sequentially, check this thread https://github.com/tonerdo/coverlet/issues/598#issuecomment-551174529

**Comment by farzadmf** [comment]
Oh wow @MarcoRossignoli , that's a good point (and a tricky one)
Just wondering: isn't the sequential (or parallel) run a potential problem with `mergeWith` anyways?

**Comment by MarcoRossignoli** [comment]
With sequential execution you'll append one result at time so at the end you'll have all results merged, very similar to https://github.com/tonerdo/coverlet/blob/master/Documentation/Examples.md#msbuild-integration

**Comment by MarcoRossignoli** [comment]
BTW the "by design" usage should be run one `dotnet test` for every test project and merge at the end, I know it not great from usage perspective, but msbuild engine doesn't handle solutions (sln) but projects(csproj) so at the moment there is no great way to synch parallel project builds, I mean when you run `dotnet test` on a solution actually you'll run one for every test projects in parallel, and there is no built-in way to `wait all tests`. That semantics is not present in msbuild.

**Comment by ckarcz** [comment]
here is a cross platform powershell script i created that finds unit test projects and generates merged coverlet format report and a final output report in cobertura format, but should work with any format. it's not perfect, but can be a good starting point for a more powerful script (checking error codes etc). it then generates a report using reportgenerator.

fwiw, i've also seen Cake integration that may do this all for you

https://gist.github.com/ckarcz/180a9ddfd9bfe33b1e68e1634c0e599e

**Comment by MarcoRossignoli** [comment]
@ckarcz doesn't https://github.com/tonerdo/coverlet/blob/master/Documentation/Examples/MSBuild/MergeWith/HowTo.md works for you?

**Comment by ckarcz** [comment]
@MarcoRossignoli that would work but i cannot get msbuild to accept the multiple values for the 
```/p:CoverletOutputFormat``` parameter when executing in powershell.

i have tired all sorts of combinations of
```/p:CoverletOutputFormat=\"cobertura,json\"```

if you know of a way that would be most excellent 👍

**Comment by ckarcz** [comment]
found a way around the issue i was having via [this msbuild issue](https://github.com/microsoft/msbuild/issues/471#issuecomment-366268743).

this is how you would need to do it in powershell:
```/p:"CoverletOutputFormat=json%2ccobertura"```

i have updated my [gist](https://gist.github.com/ckarcz/180a9ddfd9bfe33b1e68e1634c0e599e#gistcomment-3132710) with a comment with the improved version.

if you guys can update your documentation with a note for powershell users that may save some headaches! 😸

**Comment by MarcoRossignoli** [comment]
There is on documentation...but maybe too hidden https://github.com/tonerdo/coverlet/blob/master/Documentation/MSBuildIntegration.md#note-for-powershell--vsts-users

**Comment by MarcoRossignoli** [comment]
@ckarcz does it work?

**Comment by ckarcz** [comment]
@MarcoRossignoli it did in fact work, but the merge overwrites the results of a library with the coverage of the subsequent test project coverage, as per issue #203

**Comment by MarcoRossignoli** [comment]
@ckarcz asked also on the other thread can you provide a repro, also better in a new issue?
